bento/src/main/java/com/yelp/android/bento/core/ListItemTouchCallback.kt

```diff
@@ -0,0 +1,101 @@
+package com.yelp.android.bento.core
+
+import androidx.recyclerview.widget.ItemTouchHelper
+import androidx.recyclerview.widget.RecyclerView
+
+const val DRAG_FLAGS = ItemTouchHelper.UP or ItemTouchHelper.DOWN or
+        ItemTouchHelper.LEFT or ItemTouchHelper.RIGHT
+
+class ListItemTouchCallback(
+        private val component: ComponentGroup,
+        private val callback: OnItemMovedPositionListener
+) : ItemTouchHelper.Callback() {
+
+    private var dragFrom = -1
+    private var dragTo = -1
+
+    override fun canDropOver(
+            recyclerView: RecyclerView,
+            current: RecyclerView.ViewHolder,
+            target: RecyclerView.ViewHolder
+    ): Boolean {
+        // Only allow reorder if it is within the same component.
+        val fromRangeComponent = component.findReorderTargetAtIndex(current.adapterPosition)
+        val toRangeComponent = component.findReorderTargetAtIndex(target.adapterPosition)
+
+        return toRangeComponent.mValue.canDropItem(
+                fromRangeComponent.mValue,
+                current.adapterPosition - fromRangeComponent.mRange.mLower,
+                target.adapterPosition - toRangeComponent.mRange.mLower)
+    }
+
+    // Always return true here. We will check if the component is reorderable in getMovementFlags().
+    override fun isLongPressDragEnabled() = true
+
+    override fun getMovementFlags(
+            recyclerView: RecyclerView,
+            viewHolder: RecyclerView.ViewHolder
+    ): Int {
+        val lowestComponent = component.findReorderTargetAtIndex(viewHolder.adapterPosition).mValue
+        if (lowestComponent.isReorderable) {
+            if (lowestComponent is ComponentGroup) {
+                for (index in (0 until lowestComponent.size)) {
+                    if (lowestComponent[index].count != 1) {
```

Comment:
Can we use `asSequence` here?
```
if (lowestComponent.asSequence().any { it.count != 1 }) {
...
```

Also, I think we should check >0 instead of != 1 because if we have a bunch of components with a count of 0 we should be able to safely ignore them.

Reply:
I assume you mean > 1?